src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1498,11 +1498,18 @@ public static Tombstone TableValueToTombstone(JsonOperationContext context, ref
                 else
                 {
                     flags = localFlags | documentFlags;
+                    var revisionsStorage = DocumentDatabase.DocumentsStorage.RevisionsStorage;
 
+                    if (nonPersistentFlags.Contain(NonPersistentDocumentFlags.FromReplication) &&
+                        localFlags.Contain(DocumentFlags.HasRevisions) != documentFlags.Contain(DocumentFlags.HasRevisions))
+                    {
+                        var (_, count) = revisionsStorage.GetRevisions(context, id, 0, 1);
```

Comment:
We don't want that, it will materialize all the revisions.

Reply:
maybe `GetRevisionsCount` ?

Reply:
done